pkg/operator/controllers/routefix/routefix_controller.go

```diff
@@ -92,8 +92,12 @@ func (r *RouteFixReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error
 
 //SetupWithManager creates the controller
 func (r *RouteFixReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	aroClusterPredicate := predicate.NewPredicateFuncs(func(meta metav1.Object, object runtime.Object) bool {
+		return meta.GetName() == arov1alpha1.SingletonClusterName
+	})
+
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&arov1alpha1.Cluster{}).
+		For(&arov1alpha1.Cluster{}, builder.WithPredicates(aroClusterPredicate)).
 		Owns(&appsv1.DaemonSet{}).
```

Comment:
I noticed that we do not watch NS here so it is possible to change it (node selector annotation, for example). Submitted a separate, tiny PR for this: https://github.com/Azure/ARO-RP/pull/1363